uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+}
+
+M_PATH = 'uaclient.entitlements.livepatch.'  # mock path
+
+
+class TestLivepatchContractStatus(TestCase):
+
+    def test_contract_status_entitled(self):
+        """The contract_status returns ENTITLED when entitled is True."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+        self.assertEqual(status.ENTITLED, entitlement.contract_status())
+
+    def test_contract_status_unentitled(self):
+        """The contract_status returns NONE when entitled is False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        livepatch_unentitled = copy.deepcopy(LIVEPATCH_RESOURCE_ENTITLED)
+
+        # Make livepatch resource access report not entitled
+        livepatch_unentitled['entitlement']['entitled'] = False
+        cfg.write_cache('machine-access-livepatch', livepatch_unentitled)
+        entitlement = LivepatchEntitlement(cfg)
+        self.assertEqual(status.NONE, entitlement.contract_status())
+
+
+class TestLivepatchOperationalStatus(TestCase):
+
+    def test_operational_status_inapplicable_on_checked_affordances(self):
+        """The operational_status details failed check_affordances."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        livepatch_bionic = copy.deepcopy(LIVEPATCH_RESOURCE_ENTITLED)
+        livepatch_bionic['entitlement']['affordances']['series'] = ['bionic']
+        cfg.write_cache('machine-access-livepatch', livepatch_bionic)
+        entitlement = LivepatchEntitlement(cfg)
+        with mock.patch('uaclient.util.get_platform_info') as m_platform_info:
+            m_platform_info.return_value = 'xenial'
+            op_status, details = entitlement.operational_status()
+        assert op_status == status.INAPPLICABLE
+        assert 'Livepatch is not available for Ubuntu xenial.' == details
+
+    def test_contract_status_unentitled(self):
+        """The contract_status returns NONE when entitled is False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        {'entitlement': {'entitled': False}})
+        entitlement = LivepatchEntitlement(cfg)
+        self.assertEqual(status.NONE, entitlement.contract_status())
+
+
+class TestLivepatchProcessDirectives(TestCase):
+
+    @params(('remoteServer', 'remote-server=%s'), ('caCerts', 'ca-certs=%s'))
+    def test_process_directives_call_livepatch_config_command(
```

Comment:
An additional test that covers both directives being specified at once would be good.

Reply:
+1 added the multi-directive test.